### PR TITLE
Add support for percentiles aggregation

### PR DESCRIPTION
**Pull request by milicns**
Part of #118

### CODE REVIEW

File: modules/library/src/test/scala/zio/elasticsearch/ElasticAggregationSpec.scala [hunk 1]
```diff
@@ -221,6 +221,30 @@ object ElasticAggregationSpec extends ZIOSpecDefault {
...
+          ) &&
+          assert(aggregationTsRaw)(
+            equalTo(Percentiles(name = "aggregation", field = "intField.raw", Chunk.empty, missing = None))
+          ) &&
+          assert(aggregationWithMissing)(
+            equalTo(Percentiles(name = "aggregation", field = "intField", Chunk.empty, missing = Some(20.0)))
+          ) &&
+          assert(aggregationWithPercents)(
+            equalTo(Percentiles(name = "aggregation", field = "intField", Chunk(65, 90, 99), missing = None))
+          )
```
Comment (dbulaja98) [inline modules/library/src/test/scala/zio/elasticsearch/ElasticAggregationSpec.scala:246]: Make all params named.

**Review by dbulaja98 — CHANGES_REQUESTED** [review]

File: docs/overview/aggregations/elastic_aggregation_percentiles.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+val aggregationWithMissing: PercentilesAggregation = percentilesAggregation(name = "percentilesAggregation", field = Document.intField).missing(10.0)
+```
+
+If you want to add aggregation (on the same level), you can use `withAgg` method:
+```scala
+val multipleAggregations: MultipleAggregations = percentilesAggregation(name = "percentilesAggregation1", field = Document.intField).withAgg(percentilesAggregation(name = "percentilesAggregation2", field = Document.doubleField))
+```
+
+You can find more information about `Percentiles` aggregation [here](https://www.elastic.co/guide/en/elasticsearch/reference/7.17/search-aggregations-metrics-percentile-aggregation.html).
+
```
Comment (drmarjanovic) [inline docs/overview/aggregations/elastic_aggregation_percentiles.md:41]: Unnecessary new line.

File: modules/library/src/main/scala/zio/elasticsearch/aggregation/Aggregations.scala [hunk 3]
```diff
@@ -210,6 +210,48 @@ private[elasticsearch] final case class Multiple(aggregations: Chunk[SingleElast
...
+   *   an instance of the [[zio.elasticsearch.aggregation.PercentilesAggregation]] enriched with the `percents`
+   *   parameter.
+   */
+  def percents(percent: Double, percents: Double*): PercentilesAggregation
+}
+
+private[elasticsearch] final case class Percentiles(
+  name: String,
+  field: String,
+  percents: Chunk[Double],
```
Comment (drmarjanovic) [inline modules/library/src/main/scala/zio/elasticsearch/aggregation/Aggregations.scala:233]: Let's put `missing` before `percents`. I prefer alphabetically sorted fields and methods.

File: modules/library/src/main/scala/zio/elasticsearch/aggregation/Aggregations.scala [hunk 4]
```diff
@@ -210,6 +210,48 @@ private[elasticsearch] final case class Multiple(aggregations: Chunk[SingleElast
...
+  missing: Option[Double]
+) extends PercentilesAggregation { self =>
+
+  def missing(value: Double): PercentilesAggregation =
+    self.copy(missing = Some(value))
+
+  def withAgg(agg: SingleElasticAggregation): MultipleAggregations =
```
Comment (drmarjanovic) [inline modules/library/src/main/scala/zio/elasticsearch/aggregation/Aggregations.scala:240]: Please put `withAgg` below `percents`.

File: modules/library/src/main/scala/zio/elasticsearch/aggregation/Aggregations.scala [hunk 5]
```diff
@@ -210,6 +210,48 @@ private[elasticsearch] final case class Multiple(aggregations: Chunk[SingleElast
...
+    multipleAggregations.aggregations(self, agg)
+
+  def percents(percent: Double, percents: Double*): PercentilesAggregation =
```
Comment (drmarjanovic) [inline modules/library/src/main/scala/zio/elasticsearch/aggregation/Aggregations.scala:243]: Do we have any specific reason why we named this `percents` if the aggregation is `percentiles`?

**Review by drmarjanovic — CHANGES_REQUESTED** [review]

File: modules/library/src/main/scala/zio/elasticsearch/aggregation/Aggregations.scala (hunk 5 above)
Comment (milicns) [reply modules/library/src/main/scala/zio/elasticsearch/aggregation/Aggregations.scala:243]: In elastic api the parameter is named `percents`. But i can change the name to `percentiles` if it seems more intuitive.

File: docs/overview/aggregations/elastic_aggregation_percentiles.md (hunk 2 above)
Comment (dbulaja98) [reply docs/overview/aggregations/elastic_aggregation_percentiles.md:41]: We have it everywhere.

File: modules/library/src/main/scala/zio/elasticsearch/aggregation/Aggregations.scala (hunk 5 above)
Comment (milicns) [reply modules/library/src/main/scala/zio/elasticsearch/aggregation/Aggregations.scala:243]: Agreed to leave this named `percents`

**Review by dbulaja98 — APPROVED** [review]

**Pull request merged**